src/OAuth2/Storage/Pdo.php

```diff
@@ -363,18 +363,31 @@ public function scopeExists($scope)
 
     public function getDefaultScope($client_id = null)
     {
-        $stmt = $this->db->prepare(sprintf('SELECT scope FROM %s WHERE is_default=:is_default', $this->config['scope_table']));
-        $stmt->execute(array('is_default' => true));
+        $getGlobalDefaults = true;
+        $defaultScopes = array();
+
+        if (!is_null($client_id)) {
+            // Get (default) scopes from client table
+            $clientScope = $this->getClientScope($client_id);
+            if (!is_null($clientScope)) {
+                $getGlobalDefaults = false;
+                $defaultScopes = explode(' ', trim($clientScope));
```

Comment:
let's return `$clientScopes` here and remove the `$getGlobalDefaults` variable all together. This will greatly simplify this function.